src/Proxy.NET/Routes/AzureAISearch.cs

```diff
@@ -0,0 +1,62 @@
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Proxy.NET.Models;
+using Proxy.NET.Services;
+
+namespace Proxy.NET.Routes;
+
+public static class AzureAISearch
+{
+    public static RouteGroupBuilder MapAzureAISearchRoutes(this RouteGroupBuilder builder)
+    {
+        builder.MapPost("/indexes/{index}/docs/search", ProcessRequestAsync).WithMetadata(new Auth(Auth.Type.ApiKey));
+        builder.MapPost("/indexes('{index}')/docs/search.post.search", ProcessRequestAsync).WithMetadata(new Auth(Auth.Type.ApiKey));
+        return builder;
+    }
+
+    private static async Task<IResult> ProcessRequestAsync(
+        [FromServices] ICatalogService catalogService,
+        [FromServices] IProxyService proxyService,
+        [FromServices] IRequestService requestService,
+        HttpContext context,
+        string index
+    )
+    {
+        var routePattern = (context.GetEndpoint() as RouteEndpoint)?.RoutePattern.RawText;
+        var extPath = routePattern?.Split("/indexes").Last();
```

Comment:
I'm not sure I understand the reason for this? We're doing a lot of assuming non-null values or letting possible nulls slip through, so it'd be better to understand what we need here and look at how we can get it out of the route engine rather than having to parse it all ourselves.

Reply:
added a comment:  // Get the route pattern and extract the extension path that will be appended to the upstream endpoint